runtime/stdlib/test.go

```diff
@@ -1472,12 +1483,12 @@ func newMatcherWithGenericTestFunction(
 
 	inter := invocation.Interpreter
 
-	staticType, ok := testFunc.StaticType(inter).(interpreter.FunctionStaticType)
-	if !ok {
+	typeParameterPair := invocation.TypeParameterTypes.Oldest()
+	if typeParameterPair == nil {
```

Comment:
What happens if the type parameter is not provided? does it still get resolved and would be not-nil?
e.g: `newMatcher(fun (_ value: AnyStruct): Bool { ... }`

Reply:
Correct! 

For example, here's a test case for it https://github.com/onflow/cadence/pull/2130/files#diff-74ad93a1435e307865ff8474567598f410879038788126ec263e27c9244edf98R152-R154 (already existed, was just moved from the test framework to the Cadence repo in #2129)

The invocation gets the types from the elaboration, we keep them there for each invocation, and type parameters bindings are inferred from uses (here, the parameter of the test function expression passed to `newMatcher`).